lib/msal-browser/src/broker/nativeBroker/PlatformAuthProvider.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    LoggerOptions,
+    IPerformanceClient,
+    Logger,
+    AuthenticationScheme,
+} from "@azure/msal-common/browser";
+import { name, version } from "../../packageMetadata.js";
+import {
+    BrowserConfiguration,
+    DEFAULT_NATIVE_BROKER_HANDSHAKE_TIMEOUT_MS,
+} from "../../config/Configuration.js";
+import { BrowserPerformanceClient } from "../../telemetry/BrowserPerformanceClient.js";
+import { NativeMessageHandler } from "./NativeMessageHandler.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import { PlatformDOMHandler } from "./PlatformDOMHandler.js";
+
+export const PLATFORM_EXTENSION_PROVIDER = "NativeMessageHandler";
+export const PLATFORM_DOM_PROVIDER = "PlatformDOMHandler";
+
+/**
+ * Checks if the platform broker is available in the current environment.
+ * @param loggerOptions
+ * @param perfClient
+ * @returns
+ */
+export async function isPlatformBrokerAvailable(
+    loggerOptions?: LoggerOptions,
+    perfClient?: IPerformanceClient
+): Promise<boolean> {
+    const logger = new Logger(loggerOptions || {}, name, version);
+
+    logger.trace("isPlatformBrokerAvailable called");
+
+    const defaultPerformanceClientConfig = {
+        auth: {
+            clientId: "",
+        },
+    };
+
+    const performanceClient =
+        perfClient ||
+        new BrowserPerformanceClient(defaultPerformanceClientConfig);
+
+    if (!window) {
+        logger.trace("Non DOM environment detected, returning false");
+        return false;
+    }
+
+    // Check if DOM platform API is supported
+
+    // @ts-ignore
+    if (window.navigator?.platformAuthentication) {
+        const supportedContracts =
+            // @ts-ignore
+            await window.navigator.platformAuthentication.getSupportedContracts(
+                NativeConstants.MICROSOFT_ENTRA_BROKERID
+            );
+        if (supportedContracts.includes("get-token-and-sign-out")) {
+            logger.trace("Platform auth available in DOM");
+            return true;
+        }
+    }
+
+    // Check and initialize native extension provider if available
+    try {
+        const nativeExtensionProvider =
```

Comment:
If it is a breaking change ignore it, however, we need some rationalization on where we are using `native` internally and stay consistent. Not in the scope off this PR, just calling out this should be done soon.